netket/nn/symmetric_linear.py

```diff
@@ -241,10 +243,11 @@ def __call__(self, x: Array) -> Array:
         x = x.reshape(*x.shape[:3], *self.shape)
 
         x = jnp.fft.ifftn(x, s=self.shape).reshape(*x.shape[:3], self.n_cells)
-        x = x.transpose(0, 1, 3, 2).reshape(*x.shape[:2], -1)
+        x = x.transpose(0, 1, 3, 2)
+        x = x.reshape(*batch_shape, self.features, self.n_symm)
 
         if self.use_bias:
-            x += jnp.expand_dims(bias, (0, 2))
+            x += jnp.expand_dims(bias, 1)
```

Comment:
are you sure about that? it seems wrong. Before we were keeping the biases along the feature dimension, now I'm not sure. I think it will be along the 0th batch dimension.

Also, might be easier to insert this as

```python
        x = x.transpose(0, 1, 3, 2)
        x = x.reshape(-1, self.features, self.n_symm)

        if self.use_bias:
            x += jnp.expand_dims(bias, (0, 2))
        x = x.reshape(*batch_shape, self.features, self.n_symm)
```

Reply:
Broadcast dimensions are matched from the end to the front, so the nontrivial dimension maps on the 2nd to last of x in both cases.

Actually, the most elegant would be to set the `self.param` dimension of `bias` to `(self.features, 1)`, but that would likely change the numbers generated with a given PRNGKey...